control/trapezoid_velocity_profile.go

```diff
@@ -122,6 +122,8 @@ func (s *trapezoidVelocityGenerator) reset() error {
 		s.kppGain = 0.45
 	}
 
+	s.lastsetPoint = math.NaN()
```

Comment:
@randhid thoughts on this? on line 61 of this file I had previously added an additional condition `|| setPoint == 0` to the if statement because when the set point was 0, the values in this block weren't being set because `setPoint` and `lastsetPoint` both equaled 0. testing the behavior just now, I noticed that on a goTo 0 call, the motor moved extremely slow, which I realized was because this if block was always true when setPoint == 0, but the values in the block should only be calculated at the beginning of a new trap profile. by setting lastsetPoint to NaN, the original if statement behaves as necessary, but I'm not sure if we don't want to set a number to NaN, that seems hacky

Reply:
We can either do the following:

- make this a behaviour in `State` and `SetState` in the motor - where we ignore the set point if it it being set in a specific way by GoTo - which is the case you're hitting.
- merge this in and test further, I like the fact that the loop is capable of handling it - but we will have to figure out how that works with the SetVelocity loop in sensor-controlled base.

Reply:
@npmenard do you prefer the `math.Nan` approach or having the driver implementing the control package handle this case?